ci/scripts/check-bazel-tags.sh

```diff
@@ -6,14 +6,103 @@
 
 set -e
 
-untagged=$(./bazelisk.sh query "rdeps(//..., //hw:verilator) except attr(tags, '[\\[ ](verilator|manual)[,\\]]', //...)")
-if [[ ${untagged} ]]; then # Check that all targets that depend on verilator are tagged
-  echo "Target(s): ${untagged} depend(s) on //hw:verilator, please tag it with verilator or manual";
-  exit 1
-fi
+# The list of bazel tags is represented as a string and checked with a regex
+exact_regex () {
+  echo "[\\[ ]${1}[,\\]]"
+}
 
-untagged=$(./bazelisk.sh query "rdeps(//..., kind('bitstream_splice', //...)) except attr(tags, '[\\[ ](cw310_rom|cw310_test_rom|vivado|manual)[,\\]]', //...)")
-if [[ ${untagged} ]]; then # Check that all targets that depend on vivado are tagged
-  echo "Target(s): ${untagged} depend(s) on a bitstream_splice but isn't tagged with vivado or manual";
+check_empty () {
+if [[ ${1} ]]; then
+  echo "Error:"
+  echo "$1"|sed 's/^/    /';
+  echo "$2"
   exit 1
 fi
+}
+
+# This check ensures OpenTitan software can be built with a wildcard without
+# waiting for Verilator using --build_tag_filters=-verilator
+untagged=$(./bazelisk.sh query \
+  "rdeps(
+      //...,
+      //hw:verilator
+  )
+  except
+  attr(
+      tags,
+      '$(exact_regex "(verilator|manual)")',
+      //...
+  )" \
+  --output=label_kind)
+check_empty "${untagged}" \
+"Target(s) above depend(s) on //hw:verilator, please tag it with verilator or manual."
```

Comment:
Is there any guidance here on whether people should use `verilator` vs `manual`?

Reply:
This check isn't strict enough to care which is used. Neither is the one for vivado. We don't have consensus over what should be done, so we shouldn't be stricter than this. I leave it to the intent of the BUILD file's author, but I can add a note regarding why they'd choose manual.